packages/examples/react/src/app.tsx

```diff
@@ -0,0 +1,158 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2018-2022 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import 'monaco-editor/esm/vs/editor/editor.all.js';
+
+// support all editor features
+import 'monaco-editor/esm/vs/editor/standalone/browser/accessibilityHelp/accessibilityHelp.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/inspectTokens/inspectTokens.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/iPadShowKeyboard/iPadShowKeyboard.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneHelpQuickAccess.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoLineQuickAccess.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoSymbolQuickAccess.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneCommandsQuickAccess.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/quickInput/standaloneQuickInputService.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/referenceSearch/standaloneReferenceSearch.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/toggleHighContrast/toggleHighContrast.js';
+
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api.js';
+
+import { buildWorkerDefinition } from 'monaco-editor-workers';
+
+import { CloseAction, ErrorAction, MessageTransports, MonacoLanguageClient, MonacoServices } from 'monaco-languageclient';
+
+import React from 'react';
+import normalizeUrl from 'normalize-url';
+import { toSocket, WebSocketMessageReader, WebSocketMessageWriter } from 'vscode-ws-jsonrpc';
+import { StandaloneServices } from 'vscode/services';
+import getMessageServiceOverride from 'vscode/service-override/messages';
+
+StandaloneServices.initialize({
+    ...getMessageServiceOverride(document.body)
+});
+buildWorkerDefinition('dist', new URL('', window.location.href).href, false);
+
+export interface EditorProps {
+    text: string;
```

Comment:
`text` doesn't seem like a very good name, because it's only used when the component is mounted

what about `defaultText`? or `defaultCode`?

Reply:
Yes, true, I will change to `code` . As I understand it the props could be used on update (not done in this example), so no need to have the `default`prefix.

Reply:
I would expect a `code` prop to be used when updated, since it's not, it's confusing. Some big libraries in react use the `value`/`defaultValue` convention (like https://github.com/suren-atoyan/monaco-react#usage)